core/src/main/java/io/grpc/internal/ClientCallImpl.java

```diff
@@ -316,7 +316,7 @@ public void sendMessage(ReqT message) {
       InputStream messageIs = method.streamRequest(message);
       stream.writeMessage(messageIs);
     } catch (Throwable e) {
-      cancel(Status.CANCELLED.withCause(e).withDescription("Failed to stream message"));
+      stream.cancel(Status.CANCELLED.withCause(e).withDescription("Failed to stream message"));
```

Comment:
Removed TODO and cancel(Status). I sort of expect we'll add it back eventually, but it is easy to do that later if/when we need it.